pkg/backup/backupdest/backup_destination.go

```diff
@@ -657,6 +661,14 @@ func ResolveBackupManifests(
 	totalMemSize := ownedMemSize
 	ownedMemSize = 0
 
+	if !includeCompacted {
```

Comment:
i feel like this belongs in `ValidateEndTimeAndTruncate`, which does all of the manipulation of the backup chain. Furthermore, if we added it there, we could refactor `TestElideSkippedLayers` to call `ValidateEndTimeAndTruncate` to unit test this change (add an `expectedNoCompaction` field to the test case struct) instead of using another testing knob and adding more bespoke logic to our e2e tests.